File: mobile/core/domain/src/main/java/com/qrdatmon/core/domain/model/Order.kt
package com.qrdatmon.core.domain.model

data class Order(
    val id: String,
    val billId: String,
    val userId: String,
    val orderNumber: String,
    val status: String,
    val totalAmount: Double,
    val note: String? = null,
    val items: List<OrderItem> = emptyList(),
    val createdAt: String,
    val confirmedAt: String? = null,
    val cancelledAt: String? = null,
    val cancelReason: String? = null
)

data class OrderItem(
    val id: String,
    val menuItemId: String,
    val menuItemName: String,
    val menuItemImage: String? = null,
    val quantity: Int,
    val unitPrice: Double,
    val subtotal: Double,
    val note: String? = null,
    val status: String,
    val toppings: List<OrderTopping> = emptyList(),
    val startedAt: String? = null,
    val completedAt: String? = null,
    val servedAt: String? = null
)

data class OrderTopping(
    val id: String,
    val toppingId: String,
    val name: String,
    val quantity: Int = 1,
    val price: Double
)

File: web-admin/src/types/order.ts
/**
 * Order Types for QRDatMon Web Admin
 */

export type OrderStatus = 'pending' | 'confirmed' | 'preparing' | 'ready' | 'served' | 'cancelled';
export type OrderItemStatus = 'pending' | 'preparing' | 'ready' | 'served' | 'cancelled';

export interface OrderTopping {
  toppingGroupId: string;
  toppingId: string;
  name: string;
  quantity: number;
  price: number;
}

export interface OrderItem {
  id: string;
  menuItem?: {
    id: string;
    name: string;
    imageUrl?: string;
  };
  combo?: {
    id: string;
    name: string;
    imageUrl?: string;
  };
  itemName: string;
  quantity: number;
  unitPrice: number;
  subtotal: number;
  note?: string;
  status: OrderItemStatus;
  toppings: OrderTopping[];
  priority?: number;
  startedAt?: string;
  completedAt?: string;
  servedAt?: string;
  cancelledAt?: string;
  cancelReason?: string;
}

export interface Order {
  id: string;
  orderNumber: string;
  status: OrderStatus;
  totalAmount: number;
  note?: string;
  user?: {
    id: string;
    name?: string;
    email?: string;
  };
  bill?: {
    id: string;
    billNumber?: string;
    tableNumber?: string;
  };
  confirmedBy?: {
    id: string;
    name?: string;
  };
  cancelledBy?: {
    id: string;
    name?: string;
  };
  cancelReason?: string;
  items: OrderItem[];
  confirmedAt?: string;
  completedAt?: string;
  cancelledAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface OrderFilters {
  status?: OrderStatus;
  tableId?: string;
  startDate?: string;
  endDate?: string;
  search?: string;
}

export interface OrderPagination {
  page: number;
  limit: number;
  total: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPrevPage: boolean;
}
